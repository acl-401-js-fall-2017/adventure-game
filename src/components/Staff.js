import React, { Component } from 'react';
import staffPic from '../pics/staff.jpg';



export default class Staff extends Component {
  constructor() {
    super();
    this.state = {
      staffMessage: 'You won the Chocolate factory',
    };
  }

  handleClick(value, inv){
    const { removeInventory, inventory } = this.props;
    this.setState({
      message: inventory.indexOf('gobstopper') === -1 ? 'You need to give Mr Wonka a Gobstopper' : this.state.staffMessage
    },() =>  {
      if (inventory.indexOf('gobstopper') > -1) {
        removeInventory();
      }
    });
  }

  render() {
    
    return (
      <section>
        <img src={staffPic} alt="" useMap="#staffMap" />
        <map name="staffMap" id="Map">
          <area data-inv="gobstopper" alt="grid" shape="poly" coords="326,542,423,500,543,539,450,582" onClick={({ target }) => this.handleClick(target.dataset.value)} />
        </map>
        <p className="App-script">
          Place the Everlasting Gobstopper on Mr Wonka's desk.
        </p>
        <p>
          Message: {this.state.message}
        </p>
      </section>
    );
  }

}