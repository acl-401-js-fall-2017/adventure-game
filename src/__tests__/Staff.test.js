import React from 'react';

import { shallow } from 'enzyme';
import toJSON from 'enzyme-to-json';

import Staff from '../components/Staff';

describe('staff', () => {

  it('shows a Staff', () => {
    const wrapper = shallow(<Staff handleInventory={() => {}} inventory={[]} />);
    expect(toJSON(wrapper)).toMatchSnapshot();
  });
});