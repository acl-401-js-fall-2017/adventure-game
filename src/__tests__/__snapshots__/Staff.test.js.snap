// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`staff shows a Staff 1`] = `
<section
  className="challenge"
>
  <img
    alt=""
    src="staff.jpg"
    useMap="#staffMap"
  />
  <map
    id="Map"
    name="staffMap"
  >
    <area
      alt="grid"
      coords="326,542,423,500,543,539,450,582"
      data-inv="gobstopper"
      onClick={[Function]}
      shape="poly"
    />
  </map>
  <p
    className="mission"
  >
    Place the Everlasting Gobstopper on Mr Wonka's desk.
  </p>
  <p
    className="message"
  >
    Message: 
  </p>
</section>
`;
